chainer/link.py

```diff
@@ -559,6 +585,71 @@ def serialize(self, serializer):
         for name in self._persistent:
             d[name] = serializer(name, d[name])
 
+    def repeat(self, n_repeat, mode='init'):
+        """Repeat myself multiple times to make a :class:`~chainer.Sequential`.
+
+        This method returns a :class:`~chainer.Sequential` object which has
+        a same :class:`~chainer.Link` multiple times repeatedly. The ``mode``
+        argument means how to copy myself to repeat.
+
+        .. admonition:: Example
+
+            You can repeat a same link multiple times to create longer
+            :class:`~chainer.Sequential` block like this:
+
+            .. code-block:: python
+
+                class ConvBNReLU(chainer.Chain):
+
+                    def __init__(self):
+                        super(ConvBNReLU, self).__init__()
+                        with self.init_scope():
+                            self.conv = L.Convolution2D(
+                                None, 64, 3, 1, 1, nobias=True)
+                            self.bn = L.BatchNormalization(64)
+
+                    def __call__(self, x):
+                        return F.relu(self.bn(self.conv(x)))
+
+                net = ConvBNReLU().repeat(16, mode='init')
+
+            The ``net`` object contains 16 blocks, each of which is
+            ``ConvBNReLU``. And the ``mode`` was ``init``, so each block
+            is re-initialized with different parameters. If you give
+            ``copy`` to this argument, each block has same values for its
+            paramters but its object ID is different from others. If it is
+            ``share``, each block is same to others in terms of not only
+            paramters but also the object IDs because they are shallow-copied,
+            so that when the parameter of one block is changed, all the
+            parameters in the others also change.
+
+        Args:
+            n_repeat (int): Number of times to repeat.
+            mode (str): It should be either ``init``, ``copy``, or ``share``.
+                ``init`` means paramters of each repeated element in the
+                returned :class:`~chainer.Sequential` will be re-initialized,
+                so that all elements have different initial parameters.
+                ``copy`` means that the parameters will not be re-initialized
+                but object itself will be deep-copied, so that all elements
+                have same initial parameters but can be changed independently.
+                ``share`` means all the elements which construct the resulting
+                :class:`~chainer.Sequential` object are same object because
+                they are shallow-copied, so that all parameters of elements
+                share each other.
```

Comment:
are shared with each other? (maybe we need English proofreading)

Reply:
Thanks. Fixed.